proof-producer/tests/bin/proof-producer/test_zkevm_bbf_circuits.cpp

```diff
@@ -0,0 +1,100 @@
+#include <gtest/gtest.h>
+#include <memory>
+
+#include <nil/blueprint/utils/satisfiability_check.hpp>
+
+#include <nil/proof-generator/prover.hpp>
+
+class ProverTests: public ::testing::Test {
+
+    protected:
+        using CurveType = nil::crypto3::algebra::curves::pallas;
+        using HashType = nil::crypto3::hashes::keccak_1600<256>;
+        using ConstraintSystem = nil::proof_generator::Prover<CurveType, HashType>::ConstraintSystem;
+        using BlueprintFieldType = nil::proof_generator::Prover<CurveType, HashType>::BlueprintField;
+
+        static constexpr std::size_t lambda = 9;
+        static constexpr std::size_t grind = 0;
+        static constexpr std::size_t expand_factor = 2;
+        static constexpr std::size_t max_quotient_chunks = 0;
+};
+
+TEST_F(ProverTests, Bytecode) {
+    std::string trace_file_path = std::string(TEST_DATA_DIR) + "increment_multi_tx.pb";
+    nil::proof_generator::Prover<CurveType, HashType> prover(
+                        lambda,
+                        expand_factor,
+                        max_quotient_chunks,
+                        grind,
+                        "bytecode"
+    );
+
+    EXPECT_TRUE(prover.setup_prover());
+
+    EXPECT_TRUE(prover.fill_assignment_table(trace_file_path));
+
+    const auto& circuit = prover.get_constraint_system();
+    const auto& assignment_table = prover.get_assignment_table();
+
+    ASSERT_TRUE(nil::blueprint::is_satisfied<BlueprintFieldType>(circuit, assignment_table));
+}
+
+TEST_F(ProverTests, RW) {
+    std::string trace_file_path = std::string(TEST_DATA_DIR) + "increment_multi_tx.pb";
+    nil::proof_generator::Prover<CurveType, HashType> prover(
+                        lambda,
+                        expand_factor,
+                        max_quotient_chunks,
+                        grind,
+                        "rw"
+    );
+
+    EXPECT_TRUE(prover.setup_prover());
```

Comment:
IMO does not make sense to use EXPECT here, following checks rely on these setup calls

Reply:
used ASERT_TRUE